third_party/pyth/multisig-wh-message-builder/src/index.ts

```diff
@@ -0,0 +1,284 @@
+
+  const emitter = squads.getAuthorityPDA(
+    msAccount.publicKey,
+    msAccount.authorityIndex
+  );
+  console.log(`Emitter Address: ${emitter.toBase58()}`);
+
+  const newTx = await squads.createTransaction(
+    msAccount.publicKey,
+    msAccount.authorityIndex
+  );
+  console.log(`Tx Address: ${newTx.publicKey.toBase58()}`);
+
+  const message = Keypair.generate();
+  // save message to Uint8 array keypair file called mesage.json
+  fs.writeFileSync(`keys/message.json`, `[${message.secretKey.toString()}]`);
+  console.log(`Message Address: ${message.publicKey.toBase58()}`);
+
+  console.log("Creating wormhole instructions...");
+  const wormholeIxs = await getWormholeMessageIx(
+    cluster,
+    emitter,
+    emitter,
+    message.publicKey,
+    squads.connection,
+    payload
+  );
+  console.log("Wormhole instructions created.");
+
+  console.log("Creating transaction...");
+  // transfer sol to the message account
+  await squads.addInstruction(newTx.publicKey, wormholeIxs[0]);
+  // wormhole post message ix
+  await squads.addInstruction(newTx.publicKey, wormholeIxs[1]);
+
+  await squads.activateTransaction(newTx.publicKey);
+  console.log("Transaction created.");
+}
+
+async function executeMultisigTx(
+  cluster: string,
+  emitter: PublicKey,
+  walletPath: string,
+  messagePath: string,
+  txPDA: PublicKey,
+  rpcUrl: string
+) {
+  const wallet = new NodeWallet(
+    Keypair.fromSecretKey(
+      Uint8Array.from(JSON.parse(fs.readFileSync(walletPath, "ascii")))
+    )
+  );
+  console.log(`Loaded wallet with address: ${wallet.publicKey.toBase58()}`);
+
+  const message = Keypair.fromSecretKey(
+    Uint8Array.from(JSON.parse(fs.readFileSync(messagePath, "ascii")))
+  );
+  console.log(
+    `Loaded message account with address: ${message.publicKey.toBase58()}`
+  );
+
+  const squads =
+    cluster === "devnet" ? Squads.devnet(wallet) : Squads.mainnet(wallet);
+
+  const executeIx = await squads.buildExecuteTransaction(
+    txPDA,
+    wallet.publicKey
+  );
+  executeIx.keys.forEach((key) => {
+    if (key.pubkey.equals(message.publicKey)) {
+      key.isSigner = true;
+    }
+  });
+
+  // airdrop 0.1 SOL to emitter if on devnet
+  if (cluster === "devnet") {
+    const airdropSignature = await squads.connection.requestAirdrop(
+      emitter,
+      0.1 * LAMPORTS_PER_SOL
+    );
+    const { blockhash, lastValidBlockHeight } =
+      await squads.connection.getLatestBlockhash();
+    await squads.connection.confirmTransaction({
+      blockhash,
+      lastValidBlockHeight,
+      signature: airdropSignature,
+    });
+    console.log("Airdropped 0.1 SOL to emitter");
+  }
+
+  const { blockhash, lastValidBlockHeight } =
+    await squads.connection.getLatestBlockhash();
+  const executeTx = new anchor.web3.Transaction({
+    blockhash,
+    lastValidBlockHeight,
+    feePayer: wallet.payer.publicKey,
+  });
+  const provider = new anchor.AnchorProvider(squads.connection, wallet, {
```

Comment:
Using anchor here seems overkill

Reply:
We had some problems doing it without anchor provider. I think I have some ideas now.